src/control/ToolHandler.cpp

```diff
@@ -446,13 +483,20 @@ void ToolHandler::loadSettings() {
     }
 }
 
-void ToolHandler::pointCurrentToolToButtonTool() { this->currentTool = this->buttonSelectedTool; }
+void ToolHandler::pointCurrentToolToButtonTool(Button button) {
+    Tool* tool = getButtonTool(button);
+    if (this->activeTool == tool)
+        return;
+    this->activeTool = tool;
+
+    this->fireToolChanged();
+}
 
 void ToolHandler::pointCurrentToolToToolbarTool() {
-    this->currentTool = this->toolbarSelectedTool;
+    if (this->activeTool == this->toolbarSelectedTool)
+        return;
+    this->activeTool = this->toolbarSelectedTool;
 
-    this->listener->toolColorChanged(false);
-    this->listener->toolSizeChanged();
```

Comment:
These two calls are actually part of of `fireToolChanged`. Hence they are superfluous.